### PR TITLE
refactor(filters): dynamic filters part 1

### PR DESCRIPTION
**Pull request by Birkbjo**
Refactor of filter handling. Prepares support for dynamic filters. I've decided to split the component implementation in a separate PR to not make this too big.

- Adds validation for filters - previously these could be changed in URL which result in invalid values in the selectors.
- Support for other types than "string"-filters. Currently we use only string and array of strings. To prepare for future multi-selects in filters
- Moved filters out of a namespace "filter" object
   - I like the idea of all filters in one object, but it's easier to work with `useQueryParams`, since you can use builtin Param-configs for each filter.
   - If we wanted to support other types than string in one "namespaced" filter-object, we would need to write our own `encode`, `decode`. I did do this, but it got a bit complex, so I decided to go for this route.
 - Moved some filter logic and pagination stuff to `lib`. 

### Design challenges and thoughts

Originally I wanted most of the `config` of the filters to be in one object. There's multiple things that needs to be setup for a new filter, mainly these things:

1. Setup `useQueryParam`-config and validation in `filterConfig`.
2. Allow the filter to be selected for particular object: `sectionListViewsConfig`.
3. Potentially override the generated `filterQueryParam` in `parseFiltersToQueryParams`.
4. Finally implement the actual selector component.

It would be nice to have one config-object for all of these, and I did initially experiment with that. However this proved quite complex and cumbersome, mainly due to the typings. Eg. the `zod`-schema can't really be "fragmented" into a nested object. Same with the `queryParamType`-config. Also we shouldn't import `components` from `lib`, where this config would need to be.

So I decided to keep the responsibility of each part of the filter quite separate. This means "each" part is quite independent and easy to work with. However it does make it a bit fragmented to get the full overview of what you would need to do to implement a new filter.

### CODE REVIEW

File: src/lib/sectionList/filters/filterConfig.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+/* useQueryParams config-map object
+Mapping each filter to a config object that handles encoding/decoding */
+export const filterQueryParamType = {
+    [IDENTIFIABLE_KEY]: StringParam,
+    aggregationType: CustomDelimitedArrayParam,
+    domainType: CustomDelimitedArrayParam,
+    valueType: CustomDelimitedArrayParam,
+    dataSet: CustomDelimitedArrayParam,
+} as const satisfies QueryParamsConfigMap
```
Comment (Birkbjo) [inline src/lib/sectionList/filters/filterConfig.tsx:29]: I don't love the duplication here. I did experiment generating this dynamically based on the zod-schema. But it turned a bit complex and needed type castings. So this is safer.

File: src/lib/constants/sectionListView/sectionListViewsConfig.ts
Comment (Mohammer5) [inline src/lib/constants/sectionListView/sectionListViewsConfig.ts:1]: I find it hard to follow what the difference between `defaultModelViewConfig` and `modelListViewsConfig` is.

The comment above `modelListViewsConfig` suggests that any model list view config will use the default model view config although it can be overridden. But I don't see the data element model list view config in this file use the default model view config. Just from that I would expect that `defaultModelViewConfig` is internal to this file. But it's not used in this file but elsewhere.

Maybe we can reduce the amount of indirection somehow? I suppose that's something we can tackle after release as we only have two weeks until soft freeze.

File: src/lib/constants/sectionListView/viewConfigResolver.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    const { default: defaultFilters, available: defaultAvailableFilers } =
+        defaultModelViewConfig.filters
+
+    const mergedAvailableDescriptors = uniqueBy(
+        [
+            part.available || [],
+            part.overrideDefaultAvailable ? [] : defaultAvailableFilers || [],
+            part.default || [],
+        ]
+            .flat()
```
Comment (Mohammer5) [inline src/lib/constants/sectionListView/viewConfigResolver.ts:60]:
```suggestion
        [
            ...(part.available || []),
            ...(part.overrideDefaultAvailable ? [] : defaultAvailableFilers || []),
            ...(part.default || []),
        ]
```

File: src/lib/constants/sectionListView/viewConfigResolver.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            .map((propConfig) => toFilterDescriptor(propConfig)),
```
Comment (Mohammer5) [inline src/lib/constants/sectionListView/viewConfigResolver.ts:61]:
```suggestion
            .map(toFilterDescriptor),
```

File: src/lib/constants/sectionListView/sectionListViewsConfig.ts
Comment (Birkbjo) [reply src/lib/constants/sectionListView/sectionListViewsConfig.ts:1]: These are very connected. The `defaultModelViewConfig` is just to prevent duplicating the default in the `modelListViewsConfig`. These are essentially merged, as noted in the comment. Maybe renaming `defaultModelViewConfig` to `baseModelViewConfig` would make it more clear?

This is handled and merged in the `viewConfigResolver` file. This is where these are used

File: src/lib/constants/sectionListView/viewConfigResolver.ts (hunk 2 above)
Comment (Birkbjo) [reply src/lib/constants/sectionListView/viewConfigResolver.ts:60]: This seems very personal preference. I'm not a big fan of `...( condition ? true: false)` when it can be avoided (can be useful for conditionally adding keys to an object). To me `.flat` reads cleaner here.

Comment (Mohammer5) [reply src/lib/constants/sectionListView/viewConfigResolver.ts:60]: Hence the wording "Suggested change" :) Leave it as is

File: src/lib/constants/sectionListView/sectionListViewsConfig.ts
Comment (Mohammer5) [reply src/lib/constants/sectionListView/sectionListViewsConfig.ts:1]: I see. I'd like to mention that I can see a lot of complecting, but it's something we can talk about post-release

**Review by Mohammer5 — APPROVED** [review]
